lib/rust/parser/doc-parser/src/doc_sections.rs

```diff
@@ -118,22 +144,27 @@ impl DocSectionCollector {
             Some(DocSection::Paragraph { body, .. })
             | Some(DocSection::Keyed { body, .. })
             | Some(DocSection::Marked { body, .. }) => *body = text,
-            Some(DocSection::Tag { .. }) | None =>
-                self.sections.push(DocSection::Paragraph { body: text }),
+            Some(DocSection::Tag { .. })
+            | Some(DocSection::List { .. })
+            | Some(DocSection::Arguments { .. })
+            | None => self.sections.push(DocSection::Paragraph { body: text }),
```

Comment:
This is causing the empty `Paragraph` between the `List` and the "Example:" section in the new test. The empty paragraph is not needed; the newlines between sections are implicit.

```suggestion
            Some(DocSection::List { .. }) | Some(DocSection::Arguments { .. }) => (),
            Some(DocSection::Tag { .. }) | None => self.sections.push(DocSection::Paragraph { body: text }),
```